cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/BuildCommandTest.java

```diff
@@ -1323,21 +1325,17 @@ private String getNewVersionForOldDistWarning() {
     private String getOldVersionForOldDistWarning(String version) {
         SemanticVersion prevDistributionVersion = SemanticVersion.from(version);
         String prevVersionForDiagnostic;
-        if (null != prevDistributionVersion) {
-            prevVersionForDiagnostic = String.valueOf(prevDistributionVersion.minor());
-            if (prevDistributionVersion.patch() != 0) {
-                prevVersionForDiagnostic += DOT + prevDistributionVersion.patch();
-            }
-        } else {
-            prevVersionForDiagnostic = "4 or an older Update";
+        prevVersionForDiagnostic = String.valueOf(prevDistributionVersion.minor());
+        if (prevDistributionVersion.patch() != 0) {
+            prevVersionForDiagnostic += DOT + prevDistributionVersion.patch();
```

Comment:
Is there any reason for removing the null check at the beginning and the `else` block here. The `prevVersionForDiagnostic` can be null and hence I think we should leave the original logic as it is

Reply:
Actually, `prevDistributionVersion` is always not `null` which was detected by IntelliJ:
```java
public static SemanticVersion from(String versionString) {
        try {
            Version v = Version.valueOf(versionString);
            return new SemanticVersion(v);
        } catch (IllegalArgumentException e) {
            throw new ProjectException("Version cannot be empty");
        } catch (UnexpectedCharacterException e) {
            throw new ProjectException("Invalid version: '" + versionString + "'. " + e.toString());
        } catch (ParseException e) {
            throw new ProjectException("Invalid version: '" + versionString + "'. " + e.toString());
        }
    }
```
`SemanticVersion.from` always returns a new SemanticVersion or throws an exception, but never returns `null`.
![image](https://github.com/ballerina-platform/ballerina-lang/assets/29004905/86813606-38eb-47ce-850c-9afdd89d978a)